libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -245,11 +263,16 @@ static int virq_spi_add(vgic_t *vgic, struct virq_handle *virq_data)
 
 static int virq_sgi_ppi_add(vm_vcpu_t *vcpu, vgic_t *vgic, struct virq_handle *virq_data)
 {
-    if (vgic->sgi_ppi_irq[vcpu->vcpu_id][virq_data->virq] != NULL) {
-        ZF_LOGE("VIRQ %d already registered for VCPU %u\n", virq_data->virq, vcpu->vcpu_id);
+    vgic_core_t *vgic_core = get_vgic_core(vgic, vcpu->vcpu_id);
+    assert(vgic_core);
+    int virq = virq_data->virq;
+    assert((virq >= 0) && (virq <= ARRAY_SIZE(vgic_core->sgi_ppi_irq)));
```

Comment:
Minor point: this should be `<` rather than `<=` I think.

Reply:
Thanks, you are right here also. Fixed.